sdk/src/events/pollingLogProvider.ts

```diff
@@ -40,9 +44,13 @@ export class PollingLogProvider implements LogProvider {
 					this.programId,
 					this.finality,
 					undefined,
-					this.mostRecentSeenTx
+					this.mostRecentSeenTx,
+					// If skipping history, only fetch one log back, not the maximum amount available
+					skipHistory && this.firstFetch ? 1 : undefined
```

Comment:
did this save a bunch of time on the first request?

Reply:
it's less about saving time and more about having control over and knowing that the events in the event cache are the ones we've received chronologically "going forward" and not historical ones (which aren't fetched chronologically). Because the historical ones were being fetched kind of at random it screwed with our `getEventsSinceLastTxSig` method in the monorepo, which (I think correctly) assumed that all events in the cache are in order

Reply:
Because mostRecentSeenTx is undefined the first time it fetches, it will fetch a bunch of logs going backwards unless told to keep history to 1 the first time

<img width="563" alt="image" src="https://user-images.githubusercontent.com/14064888/187422811-e217498d-d0c9-45ef-9792-d0c9fa0d5326.png">